### ISSUE TITLE
Blocking Cross Origin API request issue with Jupyter Notebook.

### ISSUE THREAD

**Issue by Piraat** [jupyter/notebook#1804]
Hi there,

I'm having trouble opening my Jupyter Notebook in Chrome. It wants to open but somehow cannot connect to the Python host, see below for the command line output. Can anyone help me with this?

[I 14:56:47.921 NotebookApp] [nb_conda_kernels] enabled, 2 kernels found
[I 14:56:49.042 NotebookApp] The port 8888 is already in use, trying another por
t.
[I 14:56:49.161 NotebookApp] Ô£ô nbpresent HTML export ENABLED
[W 14:56:49.161 NotebookApp] Ô£ù nbpresent PDF export DISABLED: No module named
nbbrowserpdf.exporters.pdf
[I 14:56:49.170 NotebookApp] [nb_conda] enabled
[I 14:56:49.378 NotebookApp] [nb_anacondacloud] enabled
[I 14:56:49.657 NotebookApp] Serving notebooks from local directory: \\W2299.kpn
nl.local\homedirs01b$\kapper345
[I 14:56:49.658 NotebookApp] 0 active kernels
[I 14:56:49.658 NotebookApp] The Jupyter Notebook is running at: http://localhos
t:8889/
[I 14:56:49.660 NotebookApp] Use Control-C to stop this server and shut down all
 kernels (twice to skip confirmation).
[W 14:56:50.582 NotebookApp] Blocking Cross Origin API request.  Origin: , Host:
 localhost:8889
[W 14:56:50.595 NotebookApp] 404 GET /api/config/tree?_=1475499410431 (::1) 13.0
0ms referer=http://localhost:8889/tree
[W 14:56:50.596 NotebookApp] Blocking Cross Origin API request.  Origin: , Host:
 localhost:8889
[W 14:56:50.598 NotebookApp] 404 GET /api/config/common?_=1475499410432 (::1) 3.
00ms referer=http://localhost:8889/tree
[W 14:56:50.601 NotebookApp] Blocking Cross Origin API request.  Origin: , Host:
 localhost:8889
[W 14:56:50.602 NotebookApp] 404 GET /api/kernelspecs (::1) 2.00ms referer=http:
//localhost:8889/tree
[W 14:56:50.605 NotebookApp] Blocking Cross Origin API request.  Origin: , Host:
 localhost:8889
[W 14:56:50.608 NotebookApp] 404 GET /api/sessions?_=1475499410433 (::1) 4.00ms
referer=http://localhost:8889/tree

**Comment by minrk** [comment]
What OS and version of Chrome? Are you just trying to connect to a local notebook on your computer? It's odd that Origin would be empty (browsers don't do this, normally). Do you have any firewall / AV / proxy software running? Does it work with another browser? Does it work if you use `127.0.0.1` instead of `localhost`?

**Comment by Piraat** [comment]
Hi there,

First of all thanks for responding so fast! Answers:
- Windows 7, Chrome 53.0.2785.143 m
- Yes I'm trying to connect locally
- It doesn't work with Chrome, Firefox or Explorer
- 127.0.0.1 doesn't work either
- Looking into the firewall thing, it's a company laptop and Notebook stopped working after a software update pushed by IT, so it is very probable that it is some sort of firewall/proxy issue

**Comment by Piraat** [comment]
Update on the issue: I got it working again: apparently some extensions of Chrome were blocking the acces to the Python kernel. After deactivating the extensions, it worked again.

**Comment by aktech** [comment]
> apparently some extensions of Chrome were blocking the acces to the Python kernel.

Indeed.

**Comment by devil9967** [comment]
which extensions??

**Comment by llinfeng** [comment]
I came across this post when encountered the same error message. For my case, clearing the cache of the browser helped solve the problem. 

TL; DR.

I access my Jupyter server remotely, via a hashed password. I also kept accessing the same "session" from various locations on different browsers. At times, the workplace/session can be restored and this should have been achieved by something left in the cache.

To be more precise, as I did not find a way to only clear Jupyter/localhost-relevant cache pieces, the following two steps cleared the `Blocking Cross Origin API request` message and restored the functionality of the notebooks:
1. Launch `firefox` from the machine(Linux Mint) that hosts the Jupyter server. Visit `localhost:8889` on using the "native" Firefox browser (why not Chrome? Well, on the Linux machine, `sudo apt install firefox` got me a working Firefox, but Chrome took quite a hassle to install.)
2. Then, go back to the local machine and visit `localhost:8889`. Things should load normally thereafter.

**Comment by woshidama323** [comment]
On chrome , I stopped the extension plugin "Allow-Control-Allow-Origin" and  it works well
